packages/transport/src/modules/relationships/RelationshipsController.ts

```diff
@@ -197,7 +196,49 @@ export class RelationshipsController extends TransportController {
 
         this.eventBus.publish(new RelationshipChangedEvent(this.parent.identity.address.toString(), newRelationship));
 
-        return newRelationship;
+        return Result.ok(newRelationship);
+    }
+
+    public async canSendRelationship(parameters: ISendRelationshipParameters): Promise<Result<void>> {
+        const template = (parameters as SendRelationshipParameters).template;
+        if (!template.cache) {
+            throw this.newCacheEmptyError(RelationshipTemplate, template.id.toString());
+        }
+
+        const peerAddress = template.cache.createdBy;
+        const existingRelationshipsToPeer = await this.getExistingRelationshipsToIdentity(peerAddress);
+
+        if (existingRelationshipsToPeer.length !== 0) {
+            return Result.fail(TransportCoreErrors.relationships.relationshipCurrentlyExists(existingRelationshipsToPeer[0].status));
+        }
+
+        if (template.isExpired()) {
+            return Result.fail(TransportCoreErrors.relationships.expiredRelationshipTemplate(template.id.toString()));
+        }
+
+        const result = await this.client.canCreateRelationship(peerAddress.toString());
+
+        if (!result.value.canCreate) {
+            return Result.fail(
+                new ApplicationError(
+                    "error.transport.relationships.relationshipNotYetDecomposedByPeerOrPeerIsToBeDeleted",
+                    `A Relationship to the peer ${peerAddress} cannot be created, because the former Relationship is not yet decomposed by the peer or the peer is to be deleted.`
+                )
+            );
+        }
+
+        return Result.ok(undefined);
+    }
+
+    public async getExistingRelationshipsToIdentity(address: CoreAddress): Promise<Relationship[]> {
```

Comment:
How about `getRelationshipToIdentity` and it either returns a single Relationship or undefined? We already have this kind of return behavior in other controller get-functions. Also, in my opinion the "Existing" in the name is not necessary.

Reply:
@Magnus-Kuhn ?

Reply:
There just is a `getRelationshipToIdentity` already which is more like `getMostRecentRelationship` since it only returns the latest one but doesn't exclude rejected/revoked relationships. What would work is using that method and then checking the relationship status.